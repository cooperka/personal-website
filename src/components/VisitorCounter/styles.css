.visitor-counter {
  text-align: center;
}
