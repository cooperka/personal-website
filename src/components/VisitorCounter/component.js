// @flow

import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './styles.css';

function safeInit(appConfig, permissions) {
  if (!window.safeApp) {
    return Promise.reject('No window.safeApp');
  }

  return window.safeApp.initialise(appConfig)
    .then((appToken) => {
      console.log('Application Token received:', appToken);
      return window.safeApp.authorise(appToken, permissions)
        .then((authURI) => {
          console.log('Application was authorised by user. Auth URI received:', authURI);
          return window.safeApp.connectAuthorised(appToken, authURI)
            .then(() => {
              console.log('Application is now registered in the network.');
              return appToken;
            });
        })
        .catch((error) => {
          console.warn('Application authorisation was rejected:', error);
        });
    });
}

function safeCleanup(appToken) {
  // Make sure that SAFEApp instance is freed from memory.
  if (appToken) {
    window.safeApp.free(appToken);
    console.log('SAFEApp instance freed.');
  }
}

type Props = {
  appConfig: { id: string, name: string, vendor: string },
  permissions: { [string]: string[] },
};

type State = {
  currAppToken: ?string,
}

class VisitorCounter extends Component<Props, State> {

  static propTypes = {
    appConfig: PropTypes.object.isRequired,
    permissions: PropTypes.object.isRequired,
  };

  state = {
    currAppToken: undefined,
  };

  componentDidMount() {
    const { appConfig, permissions } = this.props;

    safeInit(appConfig, permissions)
      .then((appToken) => {
        console.log('VisitorCounter received token:', appToken);
        this.setState({ currAppToken: appToken });
      })
      .catch((error) => console.log('Could not init:', error));
  }

  componentWillUnmount() {
    safeCleanup(this.state.currAppToken);
  }

  render() {
    const { currAppToken } = this.state;

    return (
      <div className="visitor-counter">
        {`Token: ${currAppToken || 'N/A'}`}
      </div>
    );
  }

}

export default VisitorCounter;
