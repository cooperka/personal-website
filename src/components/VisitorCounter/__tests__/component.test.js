// @flow

import React from 'react';
import renderer from 'react-test-renderer';

import VisitorCounter from '../component';

const APP_CONFIG = {
  id: 'test',
  name: 'Test',
  vendor: 'cooperka',
};

const PERMISSIONS = {
  _public: ['Read'],
};

it('renders', () => {
  const tree = renderer.create(
    <VisitorCounter appConfig={APP_CONFIG} permissions={PERMISSIONS} />,
  );
  expect(tree.toJSON()).toMatchSnapshot();
});
