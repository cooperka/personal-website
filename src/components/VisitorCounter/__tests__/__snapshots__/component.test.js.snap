// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="visitor-counter"
>
  Token: N/A
</div>
`;
